pallets/identity/src/lib.rs

```diff
@@ -838,6 +851,37 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Get the usage count of a key.
+    pub fn get_account_key_usage(key: &T::AccountId) -> u64 {
+        if <AccountKeyUsage<T>>::contains_key(key) {
+            // Get a key's usage.
+            <AccountKeyUsage<T>>::get(key)
+        } else {
+            0
+        }
+    }
+
+    /// Increase the usage counter for `key`.
+    pub fn add_account_key_usage(key: &T::AccountId) {
+        // Decrease account key usage
+        <AccountKeyUsage<T>>::mutate(key, |n| *n = n.saturating_add(1_u64));
```

Comment:
Hmm... I appreciate that saturation here is a degenerate case, but perhaps we should be using `checked_add` with an error variant on overflow? I'm not entirely sure its the right move but I thought it was worth bringing up.

Reply:
No.  Throwing an error could cause a partial storage mutation.  Right now we inc/dec two account keys in the same extrinsic call.  If the first didn't overflow, but the second did, then one key would have a wrong reference count.

Reply:
Throwing an error would ofc need to take care of atomicity.